core/src/main/java/com/linecorp/armeria/server/DefaultServiceRequestContext.java

```diff
@@ -69,14 +74,14 @@
      */
     public DefaultServiceRequestContext(
             ServiceConfig cfg, Channel ch, SessionProtocol sessionProtocol,
-            String method, String path, String mappedPath, Object request,
+            String method, String path, Map<String, String> pathParams, Object request,
             @Nullable SSLSession sslSession) {
 
         super(sessionProtocol, method, path, request);
 
         this.ch = ch;
         this.cfg = cfg;
-        this.mappedPath = mappedPath;
+        this.pathParams = ImmutableMap.copyOf(pathParams);
```

Comment:
This will most likely end up with no-op since all `PathMapping` implementations return `ImmutableMap` already.